src/Controllers/SettingsController.php

```diff
@@ -105,84 +110,97 @@ public function get()
    */
   public function post(Request $request, Response $response)
   {
-    $settingMappings = null;
-    try {
-      $dataIn = $request->input('data');
+    /** @var ExternalLogs */
+    $externalLogs = pluginApp(ExternalLogs::class);
 
-      $this->logger->debug(TranslationHelper::getLoggerKey(self::LOG_KEY_CONTROLLER_IN), [
-        'additionalInfo' => ['payloadIn' => json_encode($dataIn)],
-        'method'         => __METHOD__
-      ]);
-
-      if (!isset($dataIn) || empty($dataIn)) {
-        throw new \Exception('No settings data provided');
+    try {
+      $settingMappings = null;
+      try {
+        $dataIn = $request->input('data');
+
+        $this->logger->debug(TranslationHelper::getLoggerKey(self::LOG_KEY_CONTROLLER_IN), [
+          'additionalInfo' => ['payloadIn' => json_encode($dataIn)],
+          'method'         => __METHOD__
+        ]);
+
+        if (!isset($dataIn) || empty($dataIn)) {
+          throw new \Exception('No settings data provided');
+        }
+
+        // get and validate all user input
+        $stockBuffer = self::getAndValidateStockBufferFromInput($dataIn);
+        $defaultShippingProviderId = self::getAndValidateDefaultShippingProviderFromInput($dataIn);
+        $defaultOrderStatusId = self::getAndValidateDefaultOrderStatusFromInput($dataIn);
+        $defaultItemMappingMethod = $this->getAndValidateDefaultItemMappingMethodFromInput($dataIn);
+        $importOrderSince = self::getAndValidateImportOrdersSinceFromInput($dataIn);
+        $isAllInventorySyncEnabled = self::getAndValidateAllInventorySyncFromInput($dataIn);
+
+        $settingMappings = [
+          AbstractConfigHelper::SETTINGS_STOCK_BUFFER_KEY => $stockBuffer,
+          AbstractConfigHelper::SETTINGS_DEFAULT_SHIPPING_PROVIDER_KEY => $defaultShippingProviderId,
+          AbstractConfigHelper::SETTINGS_DEFAULT_ORDER_STATUS_KEY => $defaultOrderStatusId,
+          AbstractConfigHelper::SETTINGS_DEFAULT_ITEM_MAPPING_METHOD => $defaultItemMappingMethod,
+          AbstractConfigHelper::IMPORT_ORDER_SINCE => $importOrderSince,
+          AbstractConfigHelper::SETTINGS_SEND_ALL_ITEMS_KEY => $isAllInventorySyncEnabled,
+        ];
+      } catch (\Exception $e) {
+
+        $this->logger->error(TranslationHelper::getLoggerKey(self::LOG_KEY_INVALID_SETTINGS), [
+          'additionalInfo' => [
+            'error' => $e->getMessage()
+          ],
+          'method'         => __METHOD__
+        ]);
+
+        $externalLogs->addErrorLog("Settings are invalid: " + $e->getMessage(), $e->getTraceAsString());
```

Comment:
You can just call the `logSenderService->execute` here without inserting the externalLogs and adding the finally block as this block would just end and exit.
I don't think we would ever have both exceptions happening in the externallogs.

Reply:
I agree that in general yes if we have multiple error logs and want to accumulate and send them in one shot.
In the case, if there is bad request, the saving part would not even trigger hence there is at most 1 error.
The reason I raise that there is an extra try just for the `finally` which is not necessary.
we can just 
```
try{stuff}
catch
{
send log
return
}
try{stuff}
catch{
send log
return
}
```

Reply:
alright, but the issue I have with this is that the next developer to come along could want to put a log into the method body, then their log won't be sent unless they add yet another `send log` instruction. The structure you suggest is what existed before our team took over the project, and it resulted in the loss of too many logs, because devs didn't remember to put in a `send log`, much like what I just did that caused this PR to be created.

Reply:
actually now I realize I want an external log to be sent when they change the settings, so that should prove my point...

Reply:
Good point.